lib/auth/auth_with_roles_test.go

```diff
@@ -4214,3 +4215,90 @@ func TestUnimplementedClients(t *testing.T) {
 		require.True(t, trace.IsNotImplemented(err), err)
 	})
 }
+
+func TestHeadlessAuthentication(t *testing.T) {
```

Comment:
This test tests multiple different cases, and in case of failure debugging it will be hard. Can we split the test into a few smaller ones?